packages/functional/src/pipe.ts

```diff
@@ -20,87 +18,217 @@
  *      (y) => y + 1,
  * );
  * ```
- * @param init  The initial value
- * @param fns   Any number of functions to pipe the value through
- * @returns     The final value with all functions applied
+ *
+ * @see https://github.com/ramda/ramda/blob/master/source/pipe.js
+ * @see https://github.com/darky/rocket-pipes/blob/master/index.ts
+ *
+ * @example Basic
+ * ```ts
+ * const add = (a, b) => a + b;
+ * const add10 = x => add(x, 10);
+ * const add100 = x => add(x, 100);
+ * const sum = pipe(1, add10, add100);
+ * sum === 111; // true
+ * ```
+ *
+ * @example Building a Solana transaction message
+ * ```ts
+ * const transferTransactionMessage = pipe(
+ *     // The result of the first expression...
+ *     createTransactionMessage({ version: 0 }),
+ *     // ...gets passed as the sole argument to the next function in the pipeline.
+ *     tx => setTransactionMessageFeePayer(myAddress, tx),
+ *     // The return value of that function gets passed to the next...
+ *     tx => setTransactionMessageLifetimeUsingBlockhash(latestBlockhash, tx),
+ *     // ...and so on.
+ *     tx => appendTransactionMessageInstruction(createTransferInstruction(myAddress, toAddress, amountInLamports), tx),
```

Comment:
In case you wanted this example to be runnable:

```suggestion
 *     tx => appendTransactionMessageInstruction(getTransferSolInstruction({ source, destination, amount }), tx),
```

Reply:
Ah, sorry I didn't see auto-merge was on haha.

Reply:
On that note, we should probably not auto-merge if there are unresolved conversations in the PR. Which would allow us to approve and comment without worrying about auto-merge disregarding our comments.

Reply:
#181